src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+# associated documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish, distribute,
+# sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or
+# substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+# NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+# OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# ==================================================================================================
+
+import sys
+import os
+import subprocess
+import requests
+
+__all__ = [
+    'start_nni',
+    'stop_nni',
+    'set_endpoint',
+    'version',
+    'get_experiment_status',
+    'get_experiment_profile',
+    'get_trial_job',
+    'list_trial_jobs',
+    'get_job_statistics',
+    'get_job_metrics',
+    'export_data'
+]
+
+EXPERIMENT_PATH = 'experiment'
+VERSION_PATH = 'version'
+STATUS_PATH = 'check-status'
+JOB_STATISTICS_PATH = 'job-statistics'
+TRIAL_JOBS_PATH = 'trial-jobs'
+METRICS_PATH = 'metric-data'
+EXPORT_DATA_PATH = 'export-data'
+
+API_ROOT_PATH = 'api/v1/nni'
+
+_api_endpoint = None
+
+def set_endpoint(endpoint):
+    """set endpoint of nni rest server for nnicli, for example:
+    http://localhost:8080
+    """
+    global _api_endpoint
+    _api_endpoint = endpoint
+
+def _check_endpoint():
+    if _api_endpoint is None:
+        raise AssertionError("Please call set_endpoint to specify nni endpoint")
+
+def _nni_rest_get(api_path, response_type='json'):
+    _check_endpoint()
+    uri = os.path.join(_api_endpoint, API_ROOT_PATH, api_path)
```

Comment:
Does this work on windows?

Reply:
Thanks, fixed, integration test added for Windows local mode.